docs/DesignDocs/ExtendedEnclaveInitializationData.md

```diff
@@ -0,0 +1,168 @@
+Recipients of such evidence must be able to recover and verify the original enclave identity information (including the original evidence, like MRENCLAVE and MRSIGNER in the case of SGX), based on the knowledge of the EEID, which is transmitted alongside the evidence as a new type of endorsement.
+
+
+Motivations
+----------
+
+In today's OE evidence, applications can attest their configuration as part of the `report_data` argument of `oe_get_report`.
+This mechanism works robustly when the application code is entirely and statically known to the verifier of the evidence.
+
+Things get more complicated if an enclave executes user code that is not part of the signed enclave image. For instance, consider an enclave containing a JavaScript interpreter that executes user scripts that originate from the untrusted host in shared enclave memory, and which has access to the `oe_get_report` API. It is impossible to know which script has been executed by such an enclave based on traditional evidence, even if the hash of the script is included in the `report_data`, because a malicious script can obtain valid evidence via `oe_get_report`, while pretending to be an honest script. Similarly, if an enclave loads and executes arbitrary assembly code from the host, this assembly code can use an in-enclave CPU instruction, for instance the `EREPORT` instruction on SGX, to create valid evidence - hence, it is impossible to use traditional attestation to determine which assembly code was executed by the enclave.
+
+An important instance of this problem is Azure Confidential Containers: the base enclave image contains the SGX-LKL runtime, which executes a user container. The container contains arbitrary code, including the `EREPORT` instruction to obtain valid evidence for arbitrary `report_data` – potentially impersonating other containers.
+
+One way to solve this issue is to re-compile and re-sign a new enclave image for every container we launch (with the container data being measured together with the LKL base image). However, this approach is cumbersome as we would like to be able to launch arbitrary containers on demand without having to build and sign a new enclave image every time.
+
+Another class of problems we address is dynamic attestation of the memory and threading configuration. In current OE attestation, these settings (`NumStackPage`, `NumHeapPages`, `NumTCS`) are fixed at enclave signinig time. This means that it is not possible to deploy the same enclave image with different memory and threading configurations.
+
+User Experience
+---------------
+
+The user decides to enable the use of EEID when the enclave base image is signed: while not strictly necessary, we require that EEID enclaves use `NumStackPages=0`, `NumHeapPages=0`, and `NumTCS=0`. This guarantees that enclave images meant to be used with EEID cannot be accidentally initialized with traditional attestation (the enclave would fail to load).
```

Comment:
Perhaps this can just be rephrased as "All parameters specified at enclave signature time should be set to a value that would prevent the enclave from loading, typically 0. In the case of SGX that's NumStackPages, NumHeapPages and NumTCS. This guarantees etc."